clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -609,7 +673,7 @@ public synchronized void handleCompletedBatch(ProducerBatch batch, ProduceRespon
     }
 
     public synchronized void transitionToUninitialized(RuntimeException exception) {
-        transitionTo(State.UNINITIALIZED);
+        transitionTo(State.UNINITIALIZED, exception);
```

Comment:
Did we just not have this before? And do we want a last error when we are putting back to uninitialized?

Reply:
Good catch. That was me being sloppy/assumptive. The `transitionTo` doesn't look at the incoming exception unless the target state is `FATAL_ERROR` or `ABORTABLE_ERROR`, so this was effectively a no-op anyway 🤷‍♂️. I've removed the exception from the call to `transitionTo`.